# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContext2D.cpp
# Commit: 4cca882e069d
# Full Hash: 4cca882e069d67c3a9a9064ac603b0dca23a5b39
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 21:17:53
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 2 - support vw, vh, vmin and vmax units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180619
# ==============================================================================

diff -r 2f48a6566514 -r 4cca882e069d dom/canvas/CanvasRenderingContext2D.cpp
--- a/dom/canvas/CanvasRenderingContext2D.cpp	Tue Sep 12 11:45:26 2023 +0000
+++ b/dom/canvas/CanvasRenderingContext2D.cpp	Tue Sep 12 11:45:27 2023 +0000
@@ -2820,6 +2820,10 @@
   }
   gfx::Size GetSize() const override { return Size(mSize); }
 
+  CSSSize GetCSSViewportSize() const override {
+    return GetCSSViewportSizeFromContext(mPresContext);
+  }
+
  private:
   GeckoFontMetrics GetFontMetricsForType(Type aType) const override {
     switch (aType) {